src/com/google/enterprise/adaptor/testing/RecordingResponse.java

```diff
@@ -277,12 +280,103 @@ public boolean isSecure() {
   }
 
   /**
-   * Gets an unmodifiable list of the accumulated anchors.
+   * Gets an unmodifiable multimap of the accumulated anchors.
    *
-   * @return a unmodifiable list of the accumulated anchors
+   * @return a unmodifiable multimap of the accumulated anchors
+   */
+  public AnchorMap getAnchors() {
+    return new AnchorMap(anchors);
+  }
+
+  /**
+   * An immutable multimap that supports duplicate, ordered key-value pairs.
+   * The anchor texts are the keys, and the anchor URIs are the values.
+   * <p>
+   * This class is consistent with Guava's {@code LinkedListMultimap},
+   * although it does not extend that class, or implement the
+   * {@code ListMultimap} interface or all of its methods.
    */
-  public List<Map.Entry<String, URI>> getAnchors() {
-    return unmodifiableList(anchors);
+  public static class AnchorMap {
```

Comment:
Does it make sense to make this class final? is it possibly correct to subclass it?

Reply:
The only constructor is private, so it's not possible to subclass it from outside of RecordingResponse.